### DISCUSSION TITLE
How to implement a custom favicon for login?

### DISCUSSION THREAD

**Discussion by Nickk4** [keycloak/keycloak#22642] [Q&A] [answered]
I have a custom theme implemented. The custom styling works.
I am trying to overwrite as little as possible from the original keycloak theme that I import. Therefore, I only have two files in my custom theme: `theme.properties` and `style.css`. 

I'm trying to replace the favicon with my own favicon. But it's not working.
What am I doing wrong below?

custom_theme/login/resources/public/favicon.ico
custom_theme/login/theme.properties:

```
parent=base
import=common/keycloak
favIcon=/public/favicon.ico   //I've also tried other references
favIconType=image/x-icon
styles=css/style.css
```

**Comment by ssilvert** [comment, marked as answer]
Right now these properties are only implemented for admin and account consoles.

To change the favicon in login, you will need to override the `/img/favicon.ico` file from the keycloak/common theme.

`mytheme/resources/img/favicon.ico`

**Reply by Nickk4** [reply]
Thank you, it worked!

**Reply by mahaalhaj** [reply]
can you please specify how exactly did you add the favicon?
I overwrite the favicon in common folder, and imported the common inside the login theme.properties but it still not working, any idea why? @Nickk4  @ssilvert 
Thank you in advance

**Reply by Nickk4** [reply]
I did a quick check of what I did back then. While on Keycloak 23, I have:
- `favIcon=/img/favicon.ico` at the top of my login `theme.properties` files.
- `<link rel="icon" href="${url.resourcesPath}/img/favicon.ico" />` in the head of `template.ftl` of my login theme.

Hope this helps.

**Reply by mahaalhaj** [reply]
it turns out the problem was in my icon format, and now it works, thank you!
